nhrpd/nhrp_vty.c

```diff
@@ -594,29 +595,62 @@ struct info_ctx {
 	struct vty *vty;
 	afi_t afi;
 	int count;
+	struct json_object *json;
 };
 
 static void show_ip_nhrp_cache(struct nhrp_cache *c, void *pctx)
 {
 	struct info_ctx *ctx = pctx;
 	struct vty *vty = ctx->vty;
 	char buf[2][SU_ADDRSTRLEN];
+	struct json_object *json = NULL;
 
 	if (ctx->afi != family2afi(sockunion_family(&c->remote_addr)))
 		return;
 
-	if (!ctx->count) {
+	if (ctx->json)
+		json = json_object_new_object();
```

Comment:
I think we could move this allocation under `if (json)` just replacing to `if (ctx->json)`. Not sure if I explained clearly. One IF instead of two IFs.

Reply:
I read in your mind. ok.